x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/mappings_editor.tsx

```diff
@@ -24,37 +25,78 @@ interface Props {
   indexSettings?: IndexSettings;
 }
 
+type TabName = 'fields' | 'advanced';
+
 export const MappingsEditor = React.memo(
   ({ onUpdate, defaultValue = {}, indexSettings }: Props) => {
-    const [selectedTab, selectTab] = useState<'fields' | 'advanced'>('fields');
-
-    const {
-      _source = {},
-      dynamic,
-      numeric_detection,
-      date_detection,
-      dynamic_date_formats,
-      properties = {},
-    } = defaultValue;
-
-    const { configurationDefaultValue, fieldsDefaultValue } = useMemo(
-      () => ({
-        configurationDefaultValue: {
+    const [selectedTab, selectTab] = useState<TabName>('fields');
+
+    const parsedDefaultValue = useMemo(() => {
+      const {
+        _source = {},
+        dynamic,
+        numeric_detection,
+        date_detection,
+        dynamic_date_formats,
+        properties = {},
+      } = defaultValue;
+
+      return {
+        configuration: {
           _source,
           dynamic,
           numeric_detection,
           date_detection,
           dynamic_date_formats,
         },
-        fieldsDefaultValue: properties,
-      }),
-      [defaultValue]
-    );
+        fields: properties,
+      };
+    }, [defaultValue]);
+
+    const validateConfigurationForm = async (
+      configuration: State['configuration']
+    ): Promise<{ isValid: boolean; data?: MappingsConfiguration }> => {
+      const isValid = await configuration.validate();
+
+      if (!isValid) {
+        return { isValid };
+      }
+
+      // We create a snapshot of the configuration form data.
+      const formData = configuration.data.format();
+
+      return { isValid, data: formData };
+    };
+
+    const changeTab = async (tab: TabName, [state, dispatch]: [State, Dispatch]) => {
+      if (tab === 'fields') {
+        // Navigating away from the configuration form === "sending" the form: we need to validate its data first.
+        const { isValid, data } = await validateConfigurationForm(state.configuration);
+
+        if (!isValid) {
```

Comment:
What does this logic map to in terms of UX? Are we stating here, "If the form content of the current tab isn't valid, disallow navigating to the fields tab"?

Reply:
Currently, as the form is always valid, this will never do anything. But yes, this prevents navigating to a different tab. If needed, at this stage we would show a callout to the user asking to correct the errors in the form (but not needed for now).